jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantWidget/ExtendedVariantWidget.js

```diff
@@ -31,71 +33,323 @@ export default jbrowse => {
-            propertyJSX.push(tempProp)
+            if(tempProp.length != 0){
+                propertyJSX.push(tempProp)
+            }
         }
-
         var displayJSX = []
-        for(var i = 0; i < displays.length; i++){
+        for(var i = 0; i < propertyJSX.length; i++){
             displayJSX.push(
                 <BaseCard title={displays[i].name}>
-                   <div style={{ padding: '7px', width: '100%', maxHeight: 600, overflow: 'auto' }}>
-                        <Table className={displays[i].name}>
-                            <TableHead>
-                                {propertyJSX[i]}
-                            </TableHead>
-                        </Table>
-                   </div>
+                    {propertyJSX[i]}
                 </BaseCard>
             )
         }
         return displayJSX
     }
 
+    function makeChart(samples, ref, alt, classes){
+        const [state, setState] = useState(null)
+        useEffect(() => {
+            setState(
+                <BaseCard title="Genotypes">
+                    <div>
+                    Loading genotypes...
+                    </div>
+                </BaseCard>
+            )
+            var alleleCounts = {}
+            var alleleTotal = 0
+            alleleCounts[ref] = 0
+            for(var i in alt){
+                alleleCounts[alt[i]] = 0
+            }
+            var gtCounts = {}
+            var gtTotal = 0
+            for(var sample in samples){
+                var gt = samples[sample]["GT"]
+                for(var entry in gt){
+                    // CASES
+                    // ./. -- no call
+                    if(gt[entry] == "./."){
+                        if(gtCounts["No Call"]){                          // if gtCounts entry is not null, or we have a preexisting entry for it
+                            gtCounts["No Call"] = gtCounts["No Call"] + 1 // increment count for that gt
+                            gtTotal = gtTotal + 1                         // increment our total count
+                        }
+                        else {                                   // else if gtCounts entry is null, or we don't have an entry, set to 1
+                            gtCounts["No Call"] = 1
+                            gtTotal = gtTotal + 1                // increment our total count
+                        }
+                    }
+                    // int / int
+                    else if(gt[entry]){
+                        var gtKey;                               // should be an array of len 2 after split
+                        if (gt[entry].includes("/")){            // unphased gts split on /
+                            gtKey = gt[entry].split("/")
+                        }
+                        else if(gt[entry].includes("|")){        // phased gts split on |
+                            gtKey = gt[entry].split("|")
+                        }
+                        for(var gtVal in gtKey){
```

Comment:
this is picky, but why not just make an array of alleles, which is [ref].concat(alt)? then you can just take:

gtKey[gtVal] = alleles[gtKey[gtVal]]

Reply:
Updated, could you double check the functionality is preserved with this implementation? Looks fine on my end, just want to be sure.